src/Connection.php

```diff
@@ -2002,4 +2025,15 @@ public function exec(string $sql): int
 
         return $this->executeStatement($sql);
     }
+
+    private function reconnectOnFailure()
+    {
+        try {
+            $this->executeQuery($this->getDatabasePlatform($this)->getDummySelectSQL());
+        } catch (DBALException) {
+            $this->close();
+            // Attempt to reestablish the lazy connection by sending another query.
+            $this->executeQuery($this->getDatabasePlatform($this)->getDummySelectSQL());
```

Comment:
I don't think it's necessary to do this. The connection can be reestablished by the remaining code in `connect()` (`$this->_conn = $this->_driver->connect($this->params);`)